src/app/products/components/product-list.tsx

```diff
@@ -15,65 +15,53 @@ import { formatWei } from '@/lib/utils'
   const pathname = usePathname()
   const searchParams = useSearchParams()
   const sortBy = searchParams.get('sort')
   const sortDirection = searchParams.get('dir') ?? SortDirection.DESC
 
-  async function fetchProducts() {
-    const analyticsPromises = Promise.all(
-      productTokens.map((token) =>
-        // TODO: https://github.com/IndexCoop/analytics-sdk/issues/30
-        token.address ? fetchAnalytics(token.address) : null,
-      ),
-    )
-    const apyPromises = Promise.all(
-      productTokens.map((token) =>
-        token.hasApy && token.symbol ? fetchApy(token.symbol) : null,
-      ),
-    )
-    const [analyticsResults, apyResults] = await Promise.all([
-      analyticsPromises,
-      apyPromises,
-    ])
-
-    const products = productTokens.map((token, idx) => ({
-      ...token,
-      price: analyticsResults[idx]?.navPrice,
-      delta: analyticsResults[idx]?.change24h,
-      tvl: analyticsResults[idx]?.marketCap,
-      apy:
-        apyResults[idx]?.apy !== undefined && apyResults[idx].apy !== '0'
-          ? Number(formatWei(apyResults[idx].apy))
-          : null,
-    }))
-    setProducts(sortProducts(products, sortBy, sortDirection))
-    setIsLoading(false)
-  }
-
-  useEffect(() => {
-    fetchProducts()
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
-
-  useEffect(() => {
-    const interval = setInterval(() => {
-      fetchProducts()
-    }, THIRTY_SECONDS_IN_MS)
+  const { data: products, isFetching } = useQuery({
+    initialData: [[], []] as [any[], any[]],
+    queryKey: ['product-list', sortBy, sortDirection],
+    queryFn: async () => {
+      const analyticsPromises = Promise.all(
+        productTokens.map((token) =>
+          // TODO: https://github.com/IndexCoop/analytics-sdk/issues/30
+          token.address ? fetchAnalytics(token.address) : null,
+        ),
+      )
+      const apyPromises = Promise.all(
+        productTokens.map((token) =>
+          token.hasApy && token.symbol ? fetchApy(token.symbol) : null,
+        ),
+      )
 
-    return () => clearInterval(interval)
-    // eslint-disable-next-line
-  }, [sortBy, sortDirection])
+      return Promise.all([analyticsPromises, apyPromises])
+    },
+    select: (data) => {
+      const [analyticsResults, apyResults] = data
 
-  useEffect(() => {
-    setProducts((products) => sortProducts(products, sortBy, sortDirection))
-  }, [sortBy, sortDirection])
+      return sortProducts(
+        productTokens.map((token, idx) => ({
+          ...token,
+          price: analyticsResults[idx]?.navPrice,
+          delta: analyticsResults[idx]?.change24h,
+          tvl: analyticsResults[idx]?.marketCap,
+          apy:
+            apyResults[idx]?.apy !== undefined && apyResults[idx].apy !== '0'
+              ? Number(formatWei(apyResults[idx].apy))
+              : null,
+        })) as ProductRow[],
+        sortBy ?? 'tvl',
+        sortDirection,
+      )
+    },
+    refetchInterval: THIRTY_SECONDS_IN_MS,
```

Comment:
Don't think we will need this. It's not necessary to be super up2date. As you said in your description it's enough if it reloads when the user revisits the page.